### DISCUSSION TITLE
Compiling Rosetta w/ Tensorflow - SmartFixbbSimAnnealer error

### DISCUSSION THREAD

**Discussion by amduffy** [RosettaCommons/rosetta#106] [Build & Install] [answered]
Hi all,

I'm currently trying to compile Rosetta (release 340, 2023.6) with tensorflow enabled as per [the guidelines here](https://docs.rosettacommons.org/docs/latest/scripting_documentation/RosettaScripts/Movers/movers_pages/trRosettaProtocol). However, I get an error during compilation that `src/core/pack/annealer/SmartFixbbSimAnnealer.cc:306:48: error: aggregate 'std::array<float, 35> network_input' has incomplete type and cannot be defined`. This also occurs both when compiling with and without mpi (I've attached the relevant log from the no-mpi case). Without tensorflow (i.e. extras=mpi,serialization,cxx11thread) I have been able to successfully compile Rosetta without any problems with the same compiler.

I'll also try a more recent release to see if this was addressed at some point, but I would like to stay with the same release if possible to maintain continuity with previous runs.

Any help would be greatly appreciated
[SmartFixBBAnnealer error.txt](https://github.com/user-attachments/files/16461359/SmartFixBBAnnealer.error.txt)

SCons setup:
./scons.py -j7 bin mode=release extras=cxx11thread,tensorflow (also tried with extras=mpi,serialization,cxx11thread,tensorflow using the mpi compiler)
Compiler: GCC 12.3 (MVAPICH2 w/ GCC 12.3 for mpi compilation)
OS: RHEL 9.3

**Comment by roccomoretti** [comment, marked as answer]
It may be an issue with transitive includes in the standard library -- that is, for other compilers and other conditions std::array gets brought in from the existing includes, but for the compiler you're using that's not the case and we would need to be more explicit.

Try adding the line `#include <array>`  to the top of `src/core/pack/annealer/SmartFixbbSimAnnealer.cc` -- around line 44, right after the `#include <unordered_map>` line would likely work.

**Reply by amduffy** [reply]
That seems to have done the trick in the no-mpi case - I'll try the mpi case as well to make confirm but thanks for the help!

**Reply by roccomoretti** [reply]
This should now be fixed with PR #107
